### PR TITLE
Add defaults to configuration decorator

### PR DESCRIPTION
**Pull request by iartiukhov**
Usage example:
```
@test_configuration(
    # Config options
    ['persistence_enabled', 'sync_mode', 'wal_mode', 'cache_mode', 'ssl_enabled'],
    # Possible values
    list(gen_permutations([
        [True],
        ['FULL_SYNC', 'PRIMARY_SYNC'],
        ['NONE', 'FSYNC', 'LOG_ONLY', 'BACKGROUND'],
        ['PARTITIONED', 'REPLICATED'],
        [True, False]
    ])) + list(gen_permutations(([
        [False],
        ['FULL_SYNC', 'PRIMARY_SYNC'],
        [CONFIG_NOT_APPLICABLE_OPTION],
        ['PARTITIONED', 'REPLICATED'],
        [True, False]
    ]))),
    # Defaults
    [False, 'PRIMARY_SYNC', 'LOG_ONLY', 'PARTITIONED', False]
)
```

### CODE REVIEW

File: src/tiden/configuration_decorator.py [hunk 1]
```diff
@@ -16,15 +16,19 @@
 
 from .generators import gen_permutations
 
+CONFIG_NOT_APPLICABLE_OPTION = 'CONFIG_NOT_APPLICABLE_OPTION'
+
 
 def test_configuration(*args):
```
Comment (pilshchikov) [inline src/tiden/configuration_decorator.py:22]: Can we change this with explicitly declared arguments?
I mean `(configuration_options, configurations, configuration_defaults=None)`

Comment (iartiukhov) [reply src/tiden/configuration_decorator.py:22]: No, this won't work.

Comment (iartiukhov) [reply src/tiden/configuration_decorator.py:22]: Hm, I'll try to figure out how to do this.

**Review by pilshchikov — APPROVED** [review]

**Comment by iartiukhov** [comment]
@pilshchikov I've refactored src/tiden/configuration_decorator.py. Please review again.

File: src/tiden/configuration_decorator.py (hunk 1 above)
Comment (iartiukhov) [reply src/tiden/configuration_decorator.py]: Refactored.

**Pull request merged**